### PR TITLE
Colin register checkout

### PR DESCRIPTION
**Pull request by colintalex**
-Creates test for Story 26
-Changes redirect for create order button to profile orders
-Previous tests were affected, added users to setups to comply with orders
-Adds changes to seeds file for orders

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -37,13 +37,13 @@
...
+order1.item_orders.create(order_id: order1.id, item: pull_toy, quantity: 1, price: pull_toy.price)
+order1.item_orders.create(order_id: order1.id, item: dog_bone, quantity: 2, price: dog_bone.price)
+order1.item_orders.create(order_id: order1.id, item: tire, quantity: 3, price: tire.price)
+order1.item_orders.create(order_id: order1.id, item: rim, quantity: 4, price: rim.price)
+order1.item_orders.create(order_id: order1.id, item: chew_rope, quantity: 5, price: chew_rope.price)
+order1.item_orders.create(order_id: order1.id, item: squeaky_toy, quantity: 6, price: squeaky_toy.price)
+order1.item_orders.create(order_id: order1.id, item: kong_toy, quantity: 7, price: kong_toy.price)
+order1.item_orders.create(order_id: order1.id, item: tire_parts, quantity: 8, price: tire_parts.price)
+order1.item_orders.create(order_id: order1.id, item: axle, quantity: 9, price: axle.price)
+order1.item_orders.create(order_id: order1.id, item: seat, quantity: 10, price: seat.price)
```
Comment (Maxwell-Baird) [inline db/seeds.rb:49]: Shouldn’t this be user.orders.create instead of order1.item_orders.create?

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def show
   end
 
   def create
-    order = Order.create(order_params)
+    order = current_user.orders.new(order_params)
```
Comment (colintalex) [inline app/controllers/orders_controller.rb:12]: In hindsight, this makes sense as to why we got that error from earlier

File: db/seeds.rb (hunk 1 above)
Comment (Maxwell-Baird) [reply db/seeds.rb:49]: Sorry I mean at line 38

File: app/views/user_orders/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<h1>Profile Orders</h1>
+
+<section class="orders-index">
+  <% @user.orders.each do |order| %>
```
Comment (DavidTTran) [inline app/views/user_orders/index.html.erb:4]: Think we could use current_user just for consistency as well - so we wouldn't have to pass in @user from the controller to the view

File: db/seeds.rb (hunk 1 above)
Comment (DavidTTran) [reply db/seeds.rb:49]: It's not creating an order off of the user - this is for the joins table `Item_orders` and the many to many relationship so we're setting each item_id to an order_id

File: db/seeds.rb [hunk 4]
```diff
@@ -37,13 +37,13 @@
 
 order1 = Order.create(id: 3, name: "Colin", address: "400 Wash", city: "Denver", state: "CO", zip: 80203)
```
Comment (DavidTTran) [inline db/seeds.rb:38]: This guy should be created off of a user though

File: app/views/user_orders/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    <p><%= link_to "#{order.id}", "/orders/#{order.id}" %></p>
+    <p><%= order.name %></p>
+    <p><%= order.address %></p>
+    <p><%= order.status %></p>
+    <p><%= order.created_at %></p>
```
Comment (Maxwell-Baird) [inline app/views/user_orders/index.html.erb:9]: When we get to the user story that deals with orders index, we should add some context to this variables

File: spec/features/orders/creation_spec.rb [hunk 6]
```diff
@@ -29,30 +29,37 @@
...
+      visit '/login'
 
-      fill_in :name, with: name
-      fill_in :address, with: address
-      fill_in :city, with: city
-      fill_in :state, with: state
-      fill_in :zip, with: zip
+      fill_in :email, with: user.email
+      fill_in :password, with: user.password
+      click_on 'Login'
```
Comment (DavidTTran) [inline spec/features/orders/creation_spec.rb:37]: Seems like a good spot to use a stubs. It's faster(though not to us) to just tell the program `:current_user = user` whenever we have to login

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -4,7 +4,11 @@
   describe "validations" do
     [:name, :address, :city, :state, :zip, :address, :password].each do |field|
```
Comment (DavidTTran) [inline spec/models/user_spec.rb:5]: I think if we leave this in it's going to error..

File: app/controllers/user_orders_controller.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class UserOrdersController < ApplicationController
+  def index
+    @user = User.find(session[:user_id])
```
Comment (Maxwell-Baird) [inline app/controllers/user_orders_controller.rb:3]: Do we need this? I think current_user takes care of finding who is currently login

**Pull request merged**